bolt/util.py

```diff
@@ -15,43 +20,54 @@ def get_project_root():
     return project_root
 
 
-def execute_command(command):
-    command_prepared = command_prepare(command)
-
-    print(command_prepared)
-
-    process = subprocess.run(
-        command_prepared,
-        shell=True,
-        executable='/bin/bash',
-        capture_output=True,
-        encoding='utf-8',
-    )
+def execute_command(command, log_file_path=None):
+    """
+    Executes a shell command.
 
-    if process.returncode != 0:
-        print(process)
-        print(process.stderr)
-        sys.exit(1)
+    Parameters:
+    - command: Command to be executed as a formatted string.
+    - log_file_path: Optional path to a log file to capture the command output.
 
-    return process
+    Raises:
+    - SystemExit if the command fails.
+    """
+    logger.info(command.strip())
 
+    try:
+        if log_file_path:
+            with open(log_file_path, 'w') as log_file:
+                process = subprocess.run(
+                    command,
+                    shell=True,
+                    executable='/bin/bash',
+                    check=True,
+                    stdout=log_file,
+                    stderr=subprocess.STDOUT,
+                    text=True,
+                    encoding='utf-8'
+                )
+        else:
+            process = subprocess.run(
```

Comment:
Do we get streaming of stdout / stderr from system call to terminal or log file? Or will we need another approach

Reply:
Yes we're streaming stdout / stder and get in to log file.
Unless a specified log file, mainly so multiprocess logs don't mix